### ISSUE TITLE
[rule:no-redeclare] not working in babel-eslint

### ISSUE THREAD

**Issue by anshulguleria** [babel/babel-eslint#217]
I am using `eslint v1.9.0`, `node v5.0.0` and `babel-eslint v4.1.5`.

I was trying to use `no-redeclare` rule in `eslintrc` file.
```json
{
    "parser": "babel-eslint",
    "rules": {
        "no-redeclare": [2, { "builtinGlobals": true }]
    }
}
```
And code as:
```js
let mine = 23;
let mine = 34;

console.log(mine);
```

Ideally this should give me `no-redeclare` error but it didn't.

When I removed the `parser` configuration from `eslintrc` and replaced `let` with `var` it gives me the expected error.

**Comment by hzoo** [comment]
Can you check eslint 1.10.2 and babel-eslint v5.0.0-beta4?

**Comment by anshulguleria** [comment]
@hzoo  Tried it. Still not working with latest `eslint` and `babel-eslint`

**Comment by preco21** [comment]
I have same issue.
I am using `eslint v2.5.3`, `node v5.9.1`, `babel-eslint v6.0.0`.

**Comment by danez** [comment]
This should be fixed in 6.0.4. Can anyone of you confirm for your usecase?

**Comment by anshulguleria** [comment]
Yup this seems to work fine with `eslint@2.8.0`, `babel-eslint@6.0.4` and `node@6.0.0`.

**Comment by preco21** [comment]
@danez Thanks! It works fine with `eslint@2.10.2`, `babel-eslint@6.0.4` and `node v6.2.0`.
